modules/container-definition/main.tf

```diff
@@ -57,23 +60,23 @@ locals {
     logConfiguration       = length(local.logConfiguration) > 0 ? local.logConfiguration : null
     memory                 = var.memory
     memoryReservation      = var.memoryReservation
-    mountPoints            = var.mountPoints
+    mountPoints            = var.mountPoints != null ? var.mountPoints : null
     name                   = var.name
     portMappings           = var.portMappings != null ? [for p in var.portMappings : { for k, v in p : k => v if v != null }] : null
     privileged             = local.is_not_windows ? var.privileged : null
     pseudoTerminal         = var.pseudoTerminal
     readonlyRootFilesystem = local.is_not_windows ? var.readonlyRootFilesystem : null
     repositoryCredentials  = var.repositoryCredentials
     resourceRequirements   = var.resourceRequirements
-    restartPolicy          = { for k, v in var.restartPolicy : k => v if v != null }
+    restartPolicy          = local.trimmedRestartPolicy.enabled ? local.trimmedRestartPolicy : null
```

Comment:
if `enabled = false` then we don't send any configs for `restartPolicy` to the Task Definition API call